### PR TITLE
Us 28

### PR DESCRIPTION
**Pull request by MaaaChu**
## Description
Please explain briefly the changes you made here.

## Checklist
- [x] All manual security tests passing (Described here: https://github.com/Broom94/Klattice/wiki/Manual-Security-Testing)
- [x] All unit tests passing 
- [x] All UI tests passing
- [x] Code coverage suitable
- [x] WAVE toolbar run on new or updated screens
- [x] Branch named {feature|hotfix|task}/{US-.*}
- [x] If your pull request depends on any other, please link them
- [ ] Approval from Tech lead
- [x] Approval from team member
- [x] Commit messages are meaningful
- [x] Target is green https://travis-ci.com/Broom94/Klattice/jobs/216924893 
- [x] Add `DO NOT MERGE` if you want to postpone merge (in labels)

### CODE REVIEW

File: api/app.js [hunk 1]
```diff
@@ -1,115 +1,146 @@
...
 app.listen(7999, function() {
-    console.log('Express started');
+  console.log("Express started");
 });
 
-app.get('/getroles', function(req, res) {
-    updateRoles(function(){
-        res.send(roles);
-    });
+app.get("/getroles", function(req, res) {
```
Comment (matthewdoreilly) [inline api/app.js]: Make sure you get the latest from master into your feature branch here. The naming of these endpoint URLs have changed slightly. You'll see there's a load more other merge conflicts.

File: api/db.js [hunk 2]
```diff
@@ -1,74 +1,74 @@
...
+      "SELECT user_name, user_password, user_type FROM user WHERE user_name = " +
+      "'" +
+      userName +
+      "'" +
+      " AND user_password = " +
+      "'" +
+      userPassword +
+      "'" +
+      ";";
+    db.query(queryName, function(err, rows) {
```
Comment (matthewdoreilly) [inline api/db.js]: I would have your query as a `const` at the top of this file:

`const getUserQuery = "SELECT user_name, user_password, user_type FROM user WHERE user_name = ? AND user_password = ?;`

And then change your `db.query` to:

`db.query(getUserQuery, [userName, userPassword], function(err, rows) {`

Makes it a bit more readable. Also test this for SQL injection

File: api/db.js [hunk 3]
```diff
@@ -1,74 +1,74 @@
...
+      result = -1;
+      if (err) throw err;
+      if (rows.length > 0 && rows[0].user_type == "admin") result = 1;
+      else if (rows.length > 0 && rows[0].user_type == "employee") result = 0;
+      else result = -1;
```
Comment (matthewdoreilly) [inline api/db.js]: I'll talk to yous now about this in person. But I think the code around this area could be refactored.

File: klattice/src/app/components/home/home.component.html [hunk 4]
```diff
@@ -1,166 +1,146 @@
...
-        <h1 class="familyTitle">All capabilities</h1>
+      <span aria-hidden="true" style="color: #ffffff">&times;</span>
+    </button>
+  </div>
+  <div class="modal-body">
+    <form>
+      <div class="form-group">
+        <h5 class="jobSpec">Job Specification</h5>
+        <p>{{ roleSummary }}</p>
+        <br />
```
Comment (matthewdoreilly) [inline klattice/src/app/components/home/home.component.html]: Never use `<br/>` tags. Add a class with margin or padding here.

File: klattice/src/app/components/home/home.component.ts [hunk 5]
```diff
@@ -1,39 +1,87 @@
...
+  setBandColour(name) {
+    if (name === "Executive") return "#5C068C";
+    if (name === "Leadership Community") return "#5C068C";
+    if (name === "Principal") return "#5C068C";
+    if (name === "Manager") return "#00558C";
+    if (name === "Consultant") return "#00558C";
+    if (name === "SeniorAssociate") return "#00558C";
+    if (name === "Associate") return "#009639";
+    if (name === "Trainee") return "#009639";
+    if (name === "Apprentice") return "#009639";
```
Comment (matthewdoreilly) [inline klattice/src/app/components/home/home.component.ts]: I'd change this to a switch statement (and add the strings to `const` values at the top of the file), like below:

```
switch(name) {
  case EXECUTIVE_BAND:
  case LEADERSHIP_COMMUNITY_BAND:
  case PRINCIPAL_BAND:
      return COLOUR_CONST_VARIABLE_HERE;
  case MANAGER_BAND:
  case CONSULTANT_BAND:
  case SENIOR_ASSOCIATE_BAND:
      return OTHER_COLOUR_CONST_VARIABLE_HERE;
  case ASSOCIATE_BAND:
  case TRAINEE_BAND:
  case APPRENTICE_BAND:
     return ANOTHER_COLOUR_CONST_VARIABLE_HERE;
  default:
    return SOME_DEFAULT_COLOUR_CONST_VARIABLE_HERE;
}
```

File: klattice/src/app/components/home/home.component.ts [hunk 6]
```diff
@@ -1,39 +1,87 @@
...
+  setModalHeaderColour() {
+    if (this.bandID === 1) return "#5C068C";
+    if (this.bandID === 2) return "#5C068C";
+    if (this.bandID === 3) return "#5C068C";
+    if (this.bandID === 4) return "#00558C";
+    if (this.bandID === 5) return "#00558C";
+    if (this.bandID === 6) return "#00558C";
+    if (this.bandID === 7) return "#009639";
+    if (this.bandID === 8) return "#009639";
+    if (this.bandID === 9) return "#009639";
```
Comment (matthewdoreilly) [inline klattice/src/app/components/home/home.component.ts]: See my previous comment about creating a switch statement. I would do it here as well.

**Comment by matthewdoreilly** [comment]
Make sure to update your PR checklist as you work on your ticket!

File: api/db.js (hunk 3 above)
Comment (matthewdoreilly) [reply api/db.js]: Talked to @weesteeky about the refactoring around here. Let me know if you have any questions 😄

Comment (weesteeky) [reply api/db.js]: fixed in US-029

File: klattice/src/app/components/home/home.component.ts [hunk 7]
```diff
@@ -1,39 +1,121 @@
...
+      ariaLabelledBy: "modal-basic-title"
+    });
   }
 
+  setBandColour(name) {
+    switch (name) {
+      case this.EXECUTIVE_BAND:
+      case this.LEADERSHIP_COMMUNITY_BAND:
+      case this.PRINCIPAL_BAND:
+        return "#5C068C";
```
Comment (matthewdoreilly) [inline klattice/src/app/components/home/home.component.ts]: Put these hex colours into `const`s at the top of the file as well (e.g. something like `const PURPLE_KAINOS_BRAND_COLOUR = "#5C068C";`)

**Pull request merged**